### ISSUE TITLE
Got NaN loss training with full zebra-cot dataset

### ISSUE THREAD

**Issue by wennyuhey** [multimodal-reasoning-lab/Bagel-Zebra-CoT#7]
When training the model using the provided full zebra-cot dataset, the loss often becomes NaN (and cannot converge if do not resume from an early checkpoint) during the training process. Is this common, or did I make any mistakes?

My training cmd:
`torchrun \
  --nnodes=$num_nodes \
  --node_rank=$node_rank \
  --nproc_per_node=$nproc_per_node \
  --master_addr=$master_addr \
  --master_port=$master_port \
       python train/pretrain_unified_navit.py \
          --dataset_config_file ./data/configs/zebra_cotyaml \
          --model_path $model_path \
          --layer_module Qwen2MoTDecoderLayer \
          --max_latent_size 64 \
          --resume-from $model_path \
          --finetune_from_hf True \
          --auto_resume True \
          --resume-model-only True \
          --finetune-from-ema True \
          --checkpoint-dir models/checkpoint \
          --log_every 1 \
          --lr 2e-5 \
          --lr_scheduler cosine \
          --min_lr 1e-6 \
          --num_worker 1 \
          --expected_num_tokens 60000 \
          --max_num_tokens 60000 \
          --max_num_tokens_per_sample 60000 \
          --prefer_buffer_before 30000 \
          --num_shard=$nproc_per_node \
          --sharding_strategy="FULL_SHARD" \
          --wandb_project "zebra-cot" \
          --wandb_name "h200-zebra-cot-$(date +%Y%m%d_%H%M%S)" \
          --save_every 100 \
          --warmup_steps 50 \
          --total_steps 10000 
`
and I got NaN loss frequently like the following:
`
[2025-08-07 16:37:12] (step=0000796) Train Loss mse: 0.1221, Train Loss ce: 0.5124, Train Steps/Sec: 0.02,
Yielding data with length 44006
Skipping sample due to missing image: reasoning_image_10
Yielding data with length 58978
Yielding data with length 51479
Yielding data with length 59579
Yielding data with length 56223
Yielding data with length 56148
Yielding data with length 57286
Yielding data with length 51029
Yielding data with length 49711
Yielding data with length 53997
Yielding data with length 57081
Yielding data with length 53936
Yielding data with length 59150
Yielding data with length 59033
[2025-08-07 16:37:18] Step 797: Loss is NaN, skipping this step
Yielding data with length 20309
Yielding data with length 51980
Yielding data with length 58100
Yielding data with length 22776
Yielding data with length 49108
Yielding data with length 53088
Yielding data with length 42541
Yielding data with length 54713
Yielding data with length 51515
Yielding data with length 52033
Yielding data with length 49825
Yielding data with length 56515
Yielding data with length 54545
Yielding data with length 56453
Yielding data with length 49100
Yielding data with length 56289
Yielding data with length 54317
[2025-08-07 16:37:25] Step 798: Loss is NaN, skipping this step
`
